src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -216,12 +239,11 @@ uct_ib_mlx5_devx_reg_ksm_data_addr(uct_ib_mlx5_md_t *md, struct ibv_mr *mr,
  * that were registered in single-threaded mode.
  */
 static ucs_status_t uct_ib_mlx5_devx_reg_ksm_data_contig(
```

Comment:
I have collected all the ksm registration related functions signatures and detected some inconsistency in them. E.g. all the functions took `atomic` argument after `iova` or `iova_offset` argument but `uct_ib_mlx5_devx_reg_ksm` took it right after `address`. Here is the current API aligned by arguments:

```c
uct_ib_mlx5_devx_reg_ksm            (uct_ib_mlx5_md_t *md, int atomic, uint64_t address, size_t length, int list_size, size_t entity_size, char *in,         uint32_t mkey_index, const char *reason,                                                                                 struct mlx5dv_devx_obj **mr_p, uint32_t *mkey)
uct_ib_mlx5_devx_ksm_log            (uct_ib_mlx5_md_t *md,                void *address, size_t length,            uint64_t iova,        int atomic, int mt, uint32_t mkey_index, const char *reason, ucs_status_t status)
uct_ib_mlx5_devx_reg_ksm_data_mt    (uct_ib_mlx5_md_t *md,                void *address,                           uint64_t iova,        int atomic,         uint32_t mkey_index, const char *reason, uct_ib_mlx5_devx_ksm_data_t *ksm_data,                                          struct mlx5dv_devx_obj **mr_p, uint32_t *mkey)
uct_ib_mlx5_devx_reg_ksm_data_addr  (uct_ib_mlx5_md_t *md,                void *address,                           uint64_t iova,        int atomic,         uint32_t mkey_index, const char *reason, struct mlx5dv_devx_obj **mr_p, struct ibv_mr *mr, size_t length, int list_size, struct mlx5dv_devx_obj **mr_p, uint32_t *mkey)
uct_ib_mlx5_devx_reg_ksm_data_contig(uct_ib_mlx5_md_t *md,                void *address,                           uint32_t iova_offset, int atomic,         uint32_t mkey_index, const char *reason, uct_ib_mlx5_devx_mr_t *mr,                                                      struct mlx5dv_devx_obj **mr_p, uint32_t *mkey)
uct_ib_mlx5_devx_reg_ksm_data       (uct_ib_mlx5_md_t *md, uct_ib_mlx5_devx_mem_t *memh, uct_ib_mr_type_t mr_type, uint32_t iova_offset, int atomic,         uint32_t mkey_index, const char *reason,                                                                                 struct mlx5dv_devx_obj **mr_p, uint32_t *mkey)
```

I suggest to align all the interfaces by one pattern of arguments, e.g.:
```c
uct_ib_mlx5_devx_ksm_log            (uct_ib_mlx5_md_t *md,                void *address, size_t length,            uint64_t iova,        int atomic, uint32_t mkey_index, const char *reason, int mt, ucs_status_t status)
uct_ib_mlx5_devx_reg_ksm            (uct_ib_mlx5_md_t *md,             uint64_t address, size_t length,                                  int atomic, uint32_t mkey_index, const char *reason, int list_size, size_t entity_size, char *in,                     struct mlx5dv_devx_obj **mr_p, uint32_t *mkey)
uct_ib_mlx5_devx_reg_ksm_data_mt    (uct_ib_mlx5_md_t *md,                void *address,                           uint64_t iova,        int atomic, uint32_t mkey_index, const char *reason, uct_ib_mlx5_devx_ksm_data_t *ksm_data,                           struct mlx5dv_devx_obj **mr_p, uint32_t *mkey)
uct_ib_mlx5_devx_reg_ksm_data_addr  (uct_ib_mlx5_md_t *md,                void *address, size_t length,            uint64_t iova,        int atomic, uint32_t mkey_index, const char *reason, struct mlx5dv_devx_obj **mr_p, struct ibv_mr *mr, int list_size, struct mlx5dv_devx_obj **mr_p, uint32_t *mkey)
uct_ib_mlx5_devx_reg_ksm_data_contig(uct_ib_mlx5_md_t *md,                void *address,                           uint32_t iova_offset, int atomic, uint32_t mkey_index, const char *reason, uct_ib_mlx5_devx_mr_t *mr,                                       struct mlx5dv_devx_obj **mr_p, uint32_t *mkey)
uct_ib_mlx5_devx_reg_ksm_data       (uct_ib_mlx5_md_t *md, uct_ib_mlx5_devx_mem_t *memh, uct_ib_mr_type_t mr_type, uint32_t iova_offset, int atomic, uint32_t mkey_index, const char *reason,                                                                  struct mlx5dv_devx_obj **mr_p, uint32_t *mkey)
```

In the proposed change the arguments in all the functions are placed in the following order:
1. MD
2. Info required to get address + length
3. iova or iova_offset
4. atomic
5. mkey_index
6. reason
7. MRs related info
8. unique arguments
9. devx output structure pointer
10. mkey output pointer